nimble-storage/nimble-csp-api.json

```diff
@@ -569,42 +559,52 @@
             "type" : "string"
           }
         }, {
-          "name" : "id",
-          "in" : "path",
-          "description" : "Snapshot ID",
+          "name" : "volume_id",
+          "in" : "query",
+          "description" : "Volume ID",
           "required" : true,
           "schema" : {
             "type" : "string"
           }
+        }, {
+          "name" : "name",
+          "in" : "query",
+          "description" : "Snapshot name",
+          "schema" : {
+            "type" : "string"
+          }
         } ],
         "responses" : {
           "200" : {
             "description" : "Success",
             "content" : {
               "application/json" : {
                 "schema" : {
-                  "$ref" : "#/components/schemas/ContainerVolumeSnapshot"
+                  "type" : "array",
+                  "items" : {
+                    "$ref" : "#/components/schemas/ContainerVolumeSnapshot"
+                  }
                 }
               }
             }
           },
           "400" : {
-            "description" : "Bad Request"
+            "description" : "Bad Request if volume ID is not provided"
           },
           "401" : {
             "description" : "Unauthorized"
           },
           "404" : {
-            "description" : "Not Found if snapshot ID cannot be found"
+            "description" : "Not Found if volume ID or snapshot name cannot be found"
```

Comment:
If the snapshot name isn't found, shouldn't we just return an empty list of snapshots?

I realize this only shows up in the diff because of an ordering change, so you can ignore this comment if you like.  The reordering sure makes the PR annoying to review, though.

Reply:
Ya, its annoying since we are generating these files from spec.

Reply:
At this point, we probably don't need to generate from source any longer.  Would it be a huge hassle on the nimble side to start generating stubs from the yaml from now on?

Reply:
I will look into it.